### ISSUE TITLE
eureka first config doesn't pick up config server name

### ISSUE THREAD

**Issue by brianloss** [spring-cloud/spring-cloud-config#400]
If the config server hasn't registered with eureka at the point when a eureka-first config based application is started, the application will never contact the config server. In Angel.SR6, this used to work. The user must set up retry, of course. In Angel.SR6, the DiscoveryClientConfigBootstrapConfiguration would pick up the registration (since it was listening to HeartbeatEvent events from Eureka) and call the refresh method, which would pick up the correct config server name from eureka. The retry code would then eventually retry with the corrected config server name.

It seems that commit 797544e6689138ade886217b1b4b11c231031872 removed from DiscoveryClientConfigBootstrapConfiguration the listener for HeartbeatEvent. Therefore, if a eureka-first config application launches and the config server hasn't registered yet with eureka, the refresh method on DiscoveryClientConfigBootstrapConfiguration will do nothing, and the config uri will forever be stuck at http://localhost:8888.

**Comment by dsyer** [comment]
I think you mean `DiscoveryClientConfigServiceBootstrapConfiguration`? The commit you mentioned doesn't exist, so I'm having trouble following the argument. Is it just a case of adding an additional event listener?

**Comment by brianloss** [comment]
Yes I did mean `DiscoveryClientConfigServiceBootstrapConfiguration`. Since things got moved around, the old code was in spring-cloud-netflix and that's actually where the commit is located. Sorry for the incomplete information.

As a workaround/test, I copied `DiscoveryClientConfigServiceBootstrapConfiguration`, registered my copy as a bootstrap configuration, and made it listen on `HeartbeatEvent` instead of the `ContextRefreshedEvent`. That seems to have done the trick, so I suspect just making the default `DiscoveryClientConfigServiceBootstrapConfiguration` listen on both events might be sufficient.

My use case was that I'm starting up a bunch of services in docker-compose, and my service starts before the config server is fully registered in eureka. However, another case where this could be needed is if the config server actually moves to a new location, and then someone does a refresh.
